.github/workflows/smoke.yml

```diff
@@ -17,6 +18,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - run: echo ✓
+  set_env_pr_test:
+    run: |
+      echo "SMOKE_TEST_TARGET='default'" >> $GITHUB_ENV
+    if: github.event_name == 'pull_request_target'
+  set_env_gpu_test:
+    run: |
+      echo "SMOKE_TEST_TARGET='gpu'" >> $GITHUB_ENV
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
```

Comment:
I'd even specify the following in the top level of the YAML document instead of writing those code blocks:

```yaml
env:
  SMOKE_TEST_TARGET: ${{ github.event_name == 'pull_request_target' && 'default' || 'gpu' }}
```

Reply:
Thanks!